### PR TITLE
User persistence

### PR DESCRIPTION
**Pull request by jacquelinejou**
Add user persistence so that if the user is already logged in, they don't have to relogin on launch.

### CODE REVIEW

File: Instagram/ViewController/LoginViewController.m [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+@implementation LoginViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+}
+
+- (IBAction)loginUser:(id)sender {
+    NSString *username = self.usernameField.text;
+    NSString *password = self.passwordField.text;
+    if ([self.usernameField.text isEqual:@""] || [self.passwordField.text isEqual:@""]) {
```
Comment (cychanfb) [inline Instagram/ViewController/LoginViewController.m:26]: Just FYI, isEqualToString is faster when you know that both objects are strings

File: Instagram/ViewController/LoginViewController.m [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+- (IBAction)registerUser:(id)sender {
+    // Create user
+    PFUser *newUser = [PFUser user];
+    
+    // set user properties
+    newUser.username = self.usernameField.text;
+    newUser.password = self.passwordField.text;
+    if ([self.usernameField.text isEqual:@""] || [self.passwordField.text isEqual:@""]) {
```
Comment (cychanfb) [inline Instagram/ViewController/LoginViewController.m:44]: Same here

File: Instagram/ViewController/LoginViewController.m [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        [self loginHelper];
+    } else {
+        [newUser signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * error) {
+           if (error == nil) {
+               [self performSegueWithIdentifier:@"loginSegue" sender:nil];
+           }
```
Comment (cychanfb) [inline Instagram/ViewController/LoginViewController.m:50]: I think we should have some error handling for register failed case. Is this in the assignment requirement at all?

**Comment by cychanfb** [comment]
Looks pretty good overall. Great job.

File: Instagram/ViewController/LoginViewController.m (hunk 3 above)
Comment (jacquelinejou) [reply Instagram/ViewController/LoginViewController.m:50]: There is not, there is only a log message for failure, but I will make a note to add this in.

**Pull request merged**